classes/profile.php

```diff
@@ -186,38 +194,68 @@ public static function save(\ExcimerLog $log, int $reason, int $created, float $
         $numsamples = $flamedatad3['value'];
         $flamedatad3json = json_encode($flamedatad3);
         $datasize = strlen($flamedatad3json);
-        $type = self::get_script_type();
-        $parameters = self::get_parameters($type);
-        $method = $_SERVER['REQUEST_METHOD'] ?? '';
-
-        // If set, it will trim off the leading '/' to normalise web & cli requests.
-        $request = isset($SCRIPT) ? ltrim($SCRIPT, '/') : self::REQUEST_UNKNOWN;
-        $pathinfo = $_SERVER['PATH_INFO'] ?? '';
-
-        list($contenttypevalue, $contenttypekey, $contenttypecategory) = helper::resolve_content_type($request, $pathinfo);
-
-        return $DB->insert_record('tool_excimer_profiles', [
-            'sessionid' => substr(session_id(), 0, 10),
-            'reason' => $reason,
-            'pathinfo' => $pathinfo,
-            'scripttype' => $type,
-            'userid' => $USER ? $USER->id : 0,
-            'method' => $method,
-            'created' => $created,
-            'duration' => $duration,
-            'request' => $request,
-            'parameters' => $parameters,
-            'cookies' => !defined('NO_MOODLE_COOKIES') || !NO_MOODLE_COOKIES,
-            'buffering' => !defined('NO_OUTPUT_BUFFERING') || !NO_OUTPUT_BUFFERING,
-            'responsecode' => http_response_code(),
-            'referer' => $_SERVER['HTTP_REFERER'] ?? '',
-            'datasize' => $datasize,
-            'numsamples' => $numsamples,
-            'flamedatad3' => $flamedatad3json,
-            'contenttypevalue' => $contenttypevalue,
-            'contenttypekey' => $contenttypekey,
-            'contenttypecategory' => $contenttypecategory,
-        ]);
+
+        $intrans = $DB->is_transaction_started();
+
+        if ($intrans) {
+            $cfg = $DB->export_dbconfig();
+            $db2 = \moodle_database::get_driver_instance($cfg->dbtype, $cfg->dblibrary);
+            $db2->connect($cfg->dbhost, $cfg->dbuser, $cfg->dbpass, $cfg->dbname, $cfg->prefix, $cfg->dboptions);
```

Comment:
can you add some protection is the super worst case where for some reason we can't connect. In this case just don't do the partial save and gracefully let the script continue.

This might happen when the db is getting overloaded and we max out connections and we don't want to escalate things. Please add a comment to this effect too